provider/bidengine/pricing.go

```diff
@@ -60,9 +172,175 @@ func calculatePriceRange(gspec *dtypes.GroupSpec) (sdk.Coin, sdk.Coin) {
 		cmax = rmax.Amount
 	}
 
+	if cmin.IsZero() {
+		cmin = sdk.NewInt(1)
+	}
+
 	if cmax.IsZero() {
 		cmax = sdk.NewInt(1)
 	}
 
 	return sdk.NewCoin(rmax.Denom, cmin), sdk.NewCoin(rmax.Denom, cmax)
 }
+
+type shellScriptPricing struct {
+	path         string
+	processLimit chan int
+	runtimeLimit uint
+}
+
+var errPathEmpty = errors.New("script path cannot be the empty string")
+var errProcessLimitZero = errors.New("process limit must be greater than zero")
+var errProcessRuntimeLimitZero = errors.New("process runtime limit must be greater than zero")
+
+func MakeShellScriptPricing(path string, processLimit uint, runtimeLimit uint) (BidPricingStrategy, error) {
```

Comment:
just use `time.Duration` for `runtimeLimit` from the get-go.

Reply:
you can use `time.ParseDuration` to parse the flag/env so that the input can be things like `"500ms"`.

Reply:
This change has been made